### PR TITLE
Wallet sendVtt

### PR DESCRIPTION
**Pull request by tmpolaczyk**
Follow-up to #805

Allow to send transactions previously created by the wallet. `createVtt` returns a `transactionId` which can be used to send that transaction later. API:

```json
// sendVtt method
{
  "sessionId":"c09486b1621041be7f28f8aaad7231a31cb1f188316ea2720fb7114f60f42457",
  "walletId":"631f79061a77af762dd8948b1ad7850a99425a2ff19be8f509a9a868059b6716",
  "transactionId":"39fb2dc26571f0b98ded613faf752f5e2a1cd89c5af6016519854a56217b7c5d"
}
```

Also, implemented Bech32 serialization for PKHs in the node, although there are some issues with that. So for the time being, the existing APIs that used PKHs as hex do not support Bech32, support may come in a future PR.

Also, added errors to node JSON-RPC inventory method, so for example when we try to send again a transaction that has already been confirmed (and thus the inputs are not longer valid), it will show the error message.

### CODE REVIEW

**Comment by tmpolaczyk** [comment]
Example session:

```json
> {"jsonrpc":"2.0","id":"1","method":"createWallet","params":{"seedData":"provide smoke give crop demise dance news inject behave ribbon obey confirm","seedSource":"mnemonics","password":"12345678"}}
< {"jsonrpc":"2.0","result":{"walletId":"631f79061a77af762dd8948b1ad7850a99425a2ff19be8f509a9a868059b6716"},"id":"1"}
> {"jsonrpc":"2.0","id":"1","method":"unlockWallet","params":{"walletId":"631f79061a77af762dd8948b1ad7850a99425a2ff19be8f509a9a868059b6716","password":"12345678"}}
< {"jsonrpc":"2.0","result":{"accountBalance":0,"availableAccounts":[0],"caption":null,"currentAccount":0,"name":null,"sessionExpirationSecs":3200,"sessionId":"c09486b1621041be7f28f8aaad7231a31cb1f188316ea2720fb7114f60f42457"},"id":"1"}
> {"jsonrpc":"2.0","id":"1","method":"generateAddress","params":{"sessionId":"c09486b1621041be7f28f8aaad7231a31cb1f188316ea2720fb7114f60f42457","walletId":"631f79061a77af762dd8948b1ad7850a99425a2ff19be8f509a9a868059b6716"}}
< {"jsonrpc":"2.0","result":{"address":"wit1umrmpcnyuaec2d89hpfhe68es2rdmrshhv6pty","path":"m/3'/4919'/0'/0/0"},"id":"1"}
> {"jsonrpc":"2.0","id":"1","method":"generateAddress","params":{"sessionId":"c09486b1621041be7f28f8aaad7231a31cb1f188316ea2720fb7114f60f42457","walletId":"631f79061a77af762dd8948b1ad7850a99425a2ff19be8f509a9a868059b6716"}}
< {"jsonrpc":"2.0","result":{"address":"wit12s3q6t2ned5shyzk0cn0h7eylpwhq7k3pusyda","path":"m/3'/4919'/0'/0/1"},"id":"1"}
> {"jsonrpc":"2.0","id":"1","method":"createVtt","params":{"sessionId":"c09486b1621041be7f28f8aaad7231a31cb1f188316ea2720fb7114f60f42457","walletId":"631f79061a77af762dd8948b1ad7850a99425a2ff19be8f509a9a868059b6716","address":"wit12s3q6t2ned5shyzk0cn0h7eylpwhq7k3pusyda","amount":10000,"fee":1}}
< {"jsonrpc":"2.0","error":{"code":400,"message":"Validation Error","data":[["balance","Wallet account has not enough balance"]]},"id":"1"}
> {"jsonrpc":"2.0","id":"1","method":"createVtt","params":{"sessionId":"c09486b1621041be7f28f8aaad7231a31cb1f188316ea2720fb7114f60f42457","walletId":"631f79061a77af762dd8948b1ad7850a99425a2ff19be8f509a9a868059b6716","address":"wit12s3q6t2ned5shyzk0cn0h7eylpwhq7k3pusyda","amount":1234,"fee":1}}
< {"jsonrpc":"2.0","result":{"bytes":"0ad8010a620a260a240a220a207e98ae55fa51b23e0efd7cb6b363483a2b0b85df1a7c65f941157e31af066ab8121b0a160a1454220d2d53cb690b90567e26fbfb24f85d707ad110d209121b0a160a14bd2d8bbbb2443b3476e9583d1a219d62e7a86c5a10bc4412720a4b0a490a473045022100f96efd6cf6830f118a0b8aefe37d47942740b8eacbd152fef749a7a906a556d6022046bbbdc639b523baec71666c35d423402dabca2e254da328b8a844a97e3dae1f12230a21024ab8408c8cdec561a23e09f9ffb2f11c026bacc8feb15c71f2a3956e4c1a4319","transaction":{"ValueTransfer":{"body":{"inputs":[{"output_pointer":"7e98ae55fa51b23e0efd7cb6b363483a2b0b85df1a7c65f941157e31af066ab8:0"}],"outputs":[{"pkh":"54220d2d53cb690b90567e26fbfb24f85d707ad1","value":1234},{"pkh":"bd2d8bbbb2443b3476e9583d1a219d62e7a86c5a","value":8764}]},"signatures":[{"public_key":{"bytes":[74,184,64,140,140,222,197,97,162,62,9,249,255,178,241,28,2,107,172,200,254,177,92,113,242,163,149,110,76,26,67,25],"compressed":2},"signature":{"Secp256k1":{"der":[48,69,2,33,0,249,110,253,108,246,131,15,17,138,11,138,239,227,125,71,148,39,64,184,234,203,209,82,254,247,73,167,169,6,165,86,214,2,32,70,187,189,198,57,181,35,186,236,113,102,108,53,212,35,64,45,171,202,46,37,77,163,40,184,168,68,169,126,61,174,31]}}}]}},"transactionId":"39fb2dc26571f0b98ded613faf752f5e2a1cd89c5af6016519854a56217b7c5d"},"id":"1"}
> {"jsonrpc":"2.0","id":"1","method":"sendVtt","params":{"sessionId":"c09486b1621041be7f28f8aaad7231a31cb1f188316ea2720fb7114f60f42457","walletId":"631f79061a77af762dd8948b1ad7850a99425a2ff19be8f509a9a868059b6716","transactionId":"39fb2dc26571f0b98ded613faf752f5e2a1cd89c5af6016519854a56217b7c5d"}}
< {"jsonrpc":"2.0","result":true,"id":"1"}
```

Example error message from the node:
```json
> {"jsonrpc":"2.0","id":"1","method":"sendVtt","params":{"sessionId":"c09486b1621041be7f28f8aaad7231a31cb1f188316ea2720fb7114f60f42457","walletId":"631f79061a77af762dd8948b1ad7850a99425a2ff19be8f509a9a868059b6716","transactionId":"39fb2dc26571f0b98ded613faf752f5e2a1cd89c5af6016519854a56217b7c5d"}}
< {"jsonrpc":"2.0","error":{"code":510,"message":"Node Error","data":{"cause":"request failed: RPC error: Error { code: InternalError, message: \"Output not found: 7e98ae55fa51b23e0efd7cb6b363483a2b0b85df1a7c65f941157e31af066ab8:0\", data: None }"}},"id":"1"}
```

**Comment by tmpolaczyk** [comment]
Tutorial: how to send wits to the wallet.

Use `generateAddress` wallet method, should return something like:
```json
< {"jsonrpc":"2.0","result":{"address":"wit1umrmpcnyuaec2d89hpfhe68es2rdmrshhv6pty","path":"m/3'/4919'/0'/0/0"},"id":"1"}
```

Copy that address to https://slowli.github.io/bech32-buffer/ select "Data" instead of "BtcAddress" and press decode. This should return something like e6c7b0e264e7738534e5b8537ce8f98286dd8e17.

Run CLI command

```sh
witnet node send --fee=1 --value=9999 --pkh=e6c7b0e264e7738534e5b8537ce8f98286dd8e17
```

Replacing the params as desired.

Now, to receive the transaction the wallet must be running, and the account must be unlocked. As soon as the transaction is accepted, you should see a log message in the wallet:

```
[2019-09-11T16:33:31Z INFO  witnet_wallet::repository::wallet] Found transaction to our address wit1umrmpcnyuaec2d89hpfhe68es2rdmrshhv6pty! Amount: +9999 satowits
```

And now you can use the `createVtt` wallet method.

**Comment by tmpolaczyk** [comment]
Unresolved questions:

```
// Environment for which we will format the PublicKeyHash
// Since for mainnet addresses start with "wit" but for testnet they start with "twit",
// we need this global variable to control which prefix to choose.
// An alternative would be replace all the uses of format!("{}", pkh) with
// format!("{}", pkh.bech32(environment))
// but that would force us to always carry the environment value everywhere
// Another alternative would be to supply the environment every time we create
// a PKH, but that would have more or less the same problems
```

File: wallet/src/actors/app/handlers/create_vtt.rs [hunk 1]
```diff
@@ -15,19 +21,71 @@ pub struct CreateVttRequest {
...
+        if testnet {
+            Environment::Testnet1
+        } else {
+            Environment::Mainnet
+        },
+        address,
+    )
+    .map_err(|err| {
+        log::warn!("Invalid address: {}", err);
+        app::field_error("address", "Address failed to deserialize.")
```
Comment (aesedepece) [inline wallet/src/actors/app/handlers/create_vtt.rs]:
```suggestion
        app::field_error("address", "Failed to deserialize address.")
```

**Pull request merged**